### PR TITLE
Add coremod support

### PR DESCRIPTION
**Pull request by coolsquid**
Adds coremod support. Simply add a field named "MANIFESTISTHEMOSTHORRIBLEIDEAEVER" to your manifest, with the name of an IClassTransformer class as value.

### CODE REVIEW

File: src/main/java/xyz/openmodloader/OpenModLoader.java [hunk 1]
```diff
@@ -8,33 +8,32 @@
 import xyz.openmodloader.modloader.ModLoader;
 
 public enum OpenModLoader {
-    INSTANCE;
```
Comment (shadowfacts) [inline src/main/java/xyz/openmodloader/OpenModLoader.java]: You replaced all the spaces with tabs >.>

**Comment by iLexiconn** [comment]
_alright then_

File: src/main/java/xyz/openmodloader/modloader/ModContainer.java [hunk 2]
```diff
@@ -40,20 +42,26 @@
     private String updateURL;
     @SerializedName("Logo")
     private String logoString;
+    @SerializedName("MANIFESTISTHEMOSTHORRIBLEIDEAEVER")
```
Comment (shadowfacts) [inline src/main/java/xyz/openmodloader/modloader/ModContainer.java]: pls

File: src/main/java/xyz/openmodloader/OpenModLoader.java (hunk 1 above)
Comment (iLexiconn) [inline src/main/java/xyz/openmodloader/OpenModLoader.java]: SHIT

Comment (coolsquid) [inline src/main/java/xyz/openmodloader/OpenModLoader.java]: I'm aware

File: src/main/java/xyz/openmodloader/modloader/ModLoader.java [hunk 3]
```diff
@@ -2,120 +2,116 @@
...
-import com.google.gson.JsonParser;
 
 import net.minecraft.launchwrapper.Launch;
-import net.minecraft.launchwrapper.LaunchClassLoader;
-import org.apache.commons.io.FileUtils;
 import xyz.openmodloader.OpenModLoader;
 import xyz.openmodloader.launcher.OMLAccessTransformer;
 
 public class ModLoader {
-    /**
```
Comment (shadowfacts) [inline src/main/java/xyz/openmodloader/modloader/ModLoader.java]: Bad Squidly.

**Comment by Victorious3** [comment]
I wouldn't allow coremods at all, personally. They are an annoyance

**Comment by ProfessorProspector** [comment]
Sometimes they're necessary.

**Comment by Victorious3** [comment]
@ProfessorProspector They are only necessary if the API (this) didn't serve a use case, which means that the API (this) should change in order to cover that use case.

**Comment by iLexiconn** [comment]
But we can't ever add a hook for EVERYTHING, let the modder screw with the source if they want to.

**Comment by Victorious3** [comment]
@iLexiconn Yes, yes you can. There's no such thing as an invalid *use case* as the goal is to make the entire game modable, right? Coremods are slow, ugly, and not needed. Same for access transformers in the end, but at least there is no way to use them wrongly.

**Comment by iLexiconn** [comment]
We can learn from coremods. See what they hook, and add the hooks ourselves. Like that we can slowly remove coremods and everyone's happy.

**Comment by skyem123** [comment]
How about forcing coremods to (somehow) not be normal mods? Like, they don't have access to most APIs, so they can only provide hooks? Not too sure if it's possible, though

**Comment by iLexiconn** [comment]
_but why_

**Comment by Victorious3** [comment]
@iLexiconn Uh yeah, provided you ever notice. People are lazy in that regards, don't encourage them to get away with it...

**Pull request merged**